.changeset/dry-tables-sniff.md

```diff
@@ -0,0 +1,62 @@
+---
+'@backstage/plugin-catalog-backend-module-azure': patch
+---
+
+Add a new provider `AzureDevOpsEntityProvider` as replacement for `AzureDevOpsDiscoveryProcessor`.
+
+In order to migrate from the `AzureDevOpsDiscoveryProcessor` you need to apply
+the following changes:
+
+**Before:**
+
+```yaml
+# app-config.yaml
+
+catalog:
+  locations:
+    - type: azure-discovery
+      target: https://dev.azure.com/myorg/myproject
+```
+
+```ts
+/* packages/backend/src/plugins/catalog.ts */
+
+import { AzureDevOpsDiscoveryProcessor } from '@backstage/plugin-catalog-backend-module-azure';
+
+const builder = await CatalogBuilder.create(env);
+/** ... other processors ... */
+builder.addProcessor(new AzureDevOpsDiscoveryProcessor(env.reader));
+```
+
+**After:**
+
+```yaml
+# app-config.yaml
+
+catalog:
+  providers:
+    azureDevOps:
+      anyProviderId:
+        organization: myorg
```

Comment:
Looking at this at first I didn't think it would work for me as an on-premise users. Could we expand this to include a more full example? Let's say:

```yaml
anyProviderId:
  host: selfhostedazure.yourcompany.com # This is only really needed for on-premise user
  organization: myorg # For on-premise this would be your Collection
  project: myproject
  repository: service
  path: /catalog-info.yaml
```

I know you link to the detailed docs but when I'm doing an upgrade I find having more details here super helpful and reduces having to jump around

Reply:
Done!